transform-plugins/src/main/java/co/cask/hydrator/plugin/CSVParser.java

```diff
@@ -176,26 +176,26 @@ public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) th
       org.apache.commons.csv.CSVParser parser = org.apache.commons.csv.CSVParser.parse(body, csvFormat);
       List<CSVRecord> records = parser.getRecords();
       for (CSVRecord record : records) {
-        if (fields.size() == record.size()) {
-          StructuredRecord sRecord = createStructuredRecord(record);
-          emitter.emit(sRecord);
-        } else {
-          LOG.warn("Skipping record as output schema specified has '{}' fields, while CSV record has '{}'",
-                   fields.size(), record.size());
-          // Write the record to error Dataset.
-        }
+        emitter.emit(createStructuredRecord(record, in));
       }
     } catch (IOException e) {
-      LOG.error("There was a issue parsing the record. ", e.getLocalizedMessage());
+      throw new RuntimeException("There was a issue parsing the record. " + e.getLocalizedMessage());
```

Comment:
Preserves the original exception cause when rethrowing exception. Also, better use `throw Throwables.propagate(e);` in this case.